gips/data/gpm/gpm.py

```diff
@@ -0,0 +1,335 @@
+            'description': 'Daily Accumulated Precipitation (in mm) at 23:59 UTC at 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        'IMERG-DAY-EARLY': {
+            'host': "jsimpson.pps.eosdis.nasa.gov",
+            'pattern': r'3B-HHR-L\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 06:59 UTC 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        'IMERG-DAY-MID': {
+            'host': "jsimpson.pps.eosdis.nasa.gov",
+            'pattern': r'3B-HHR-L\.MS\.MRG\.3IMERG\..{8}-S.{6}-E.{6}\..{4}\..{4}\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 14:59 0.1 degrees - Production Run '
+                           'using the IMERG Algorithm developed for the GPM constellation',
+            'path': '/NRTPUB/imerg/gis/',
+            'startdate': datetime.date(2014, 3, 12),
+            'latency': 1,
+        },
+        '3B42-DAY-LATE': {
+            'host': "trmmopen.pps.eosdis.nasa.gov",
+            'pattern': r'3B42RT\..{10}[\.7]*\.1day.tif',
+            'description': 'Daily Accumulated Precipitation (in mm) at 0.25 degrees - Production Run using the '
+                           '3B42 Algorithm developed for the TRMM constellation',
+            'path': '/pub/gis/',
+            'startdate': datetime.date(2000, 3, 2),
+            'latency': 1,
+        },
+    }
+
+    def __init__(self, filename):
+        """ Inspect a single file and get some metadata """
+        super(gpmAsset, self).__init__(filename)
+
+        bname = os.path.basename(filename)
+        try:
+            indicator = (re.search('(?<=3B-)\w*(?=-{1})', bname)).group(0)
+        except AttributeError:
+            indicator = (re.search('(?<=3B)\w*(?=\.{1})', bname)).group(0)
+            pass
+
+        if indicator == 'DAY':
+            self.asset = 'IMERG-DAY-FINAL'
+        elif indicator == 'HHR':
+            self.asset = 'IMERG-DAY-LATE'
+        elif indicator == '42RT':
+            self.asset = '3B42-DAY-LATE'
+
+        # Will break in the next millenium
+        date_here = (re.search('2[0-9]{7}', bname)).group(0)
+        self.date = datetime.datetime.strptime(date_here, "%Y%m%d").date()
+        try:
+            self._version = (re.search('V[0-9A-Z]*', bname)).group(0)
+        except AttributeError:
+            self._version = 7 #hard code for 3B42 products
+            pass
+
+        self.tile = 'h01v01'
+
+    @classmethod
+    def ftp_connect(cls, asset, date):
```

Comment:
be aware that we have a branch that might merge that may change the ftp code somewhat.  Because that might merge before this branch does, it's okay to leave this the way it is for now.